sunpy/map/maputils.py

```diff
@@ -375,3 +375,29 @@ def on_disk_bounding_coordinates(smap):
     return SkyCoord([np.nanmin(tx), np.nanmax(tx)] * u.arcsec,
                     [np.nanmin(ty), np.nanmax(ty)] * u.arcsec,
                     frame=smap.coordinate_frame)
+
+
+def contains_coordinate(smap, coordinates):
+    """
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    coordinates : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frames.Helioprojective`
+        The input coordinate. The coordinate frame must be
+        `~sunpy.coordinates.Helioprojective`.
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` If coordinates are within the coordinate
+        bound of the smap and Returns `False` If coordinates are
+        not within the coordinate bound of the smap.
+    """
+    _verify_coordinate_helioprojective(smap.coordinate_frame)
+    # Getting pixel indices of smap
+    xs, ys = all_pixel_indices_from_map(smap)
```

Comment:
You don't need to calculate all the pixel indices, just calculate the pixel position of the input coordinates and ensure that the results are within the size of `smap.dimensions`.